src/main/java/org/broadinstitute/consent/http/resources/DatasetResource.java

```diff
@@ -418,9 +420,10 @@ public Response indexDataset(@PathParam("datasetId") Integer datasetId) {
   @DELETE
   @Path("/index/{datasetId}")
   @RolesAllowed(ADMIN)
-  public Response deleteDatasetIndex(@PathParam("datasetId") Integer datasetId) {
+  public Response deleteDatasetIndex(@Auth AuthUser authUser, @PathParam("datasetId") Integer datasetId) {
```

Comment:
What is the use case of this API? Since the code is using ad-hoc operations to index datasets, I don't see how this will prevent the index from being created on another operation.

Reply:
The use case here it to ensure that a specific dataset exists in the index. We have needed this as an admin when we wanted to re-index a dataset for some reason. This will NOT prevent the index from being updated in other operations, which is intentional - we explicitly DO need to update the index in other operations.

Reply:
> ensure that a specific dataset exists in the index

Do you mean does not exist? This API deletes a dataset from the index.

My point before was that this wasn't setting last index date to null, so it wouldn't prevent it from being re-indexed. Now it does set it to null, so that works. But I'm still wondering why we'd want to delete a specific dataset from the index.

Reply:
> > ensure that a specific dataset exists in the index
> 
> Do you mean does not exist? This API deletes a dataset from the index.
> 

Thank you, yes, I did mean that it is intended to remove an entry from the index.

> But I'm still wondering why we'd want to delete a specific dataset from the index.

We have had requests from Jonathan to remove a specific dataset when the custodian has specifically requested it. Those are always one-off requests - eventually the dataset gets re-indexed when the custodian is ready for it.